packages/odie-client/src/utils/chat-utils.ts

```diff
@@ -0,0 +1,19 @@
+import { getTimestamp } from './get-timestamp';
+import type { Chat } from '../types';
+
+const MAX_ESCALATION_ATTEMPT_TIME = 3 * 24 * 60 * 60 * 1000; // three days
+
+export const hasRecentEscalationAttempt = ( chat: Chat ) => {
+	return (
+		chat?.messages?.some( ( message ) => {
+			if ( ! message.context?.flags?.forward_to_human_support || ! message.created_at ) {
+				return false;
+			}
+
+			const messageTimestamp = getTimestamp( message.created_at ) * 1000;
+			const threeDaysAgo = Date.now() - MAX_ESCALATION_ATTEMPT_TIME;
+
+			return messageTimestamp >= threeDaysAgo;
+		} ) ?? false
```

Comment:
We could optimize this a bit. Right now we iterate over all chat messages and check timestamps inside the loop, but since we know the valid window (last 3 days), we could limit the iteration to only messages within that time range. That should reduce unnecessary checks on long conversations.